runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkJobServerDriver.java

```diff
@@ -76,6 +79,10 @@ public static void main(String[] args) {
       printUsage(parser);
       return;
     }
+    //TODO: Expose the fileSystem related options.
+    // Register standard file systems.
+    FileSystems.setDefaultPipelineOptions(
```

Comment:
What's the purpose of these lines? Are the standard filesystems (`file:` and `gs:` in this case) not registered without this call? If so, is there a way to achieve the same effect without instantiating a dummy `PipelineOptions` (especially one that has been constructed from an empty `Struct` proto)?

Reply:
PipelineOptions allows providing the credentials for the filesystem. Really this should be something like PipelineOptionsFactory.fromArgs(args).

Reply:
We could get rid of the custom command line parser and replace it with PipelineOptions :)

Reply:
Absolutely, we can start using PipelineOptions instead of command line parser.
However this also raises the question on what parameters should be expected from jobserver and what parameters from pipeline. 
I think with the portability framework, all the arguments should be provided by job server which will be merged-overwritten by the pipeline options.

Reply:
Maybe we could use fromArgs() on the args remaining after parsing with the custom parser.
The jobserver parameters and pipeline options are entirely separate I think, and I can't think of a situation where they would need to be merged, even if both are represented by the PipelineOptions class.

That said, for now I am fine with just using PipelineOptionsFactory.create() (you don't need an empty struct) and fixing it when we sort out credentials better.